linenoise.c

```diff
@@ -1168,7 +1170,11 @@ int linenoiseHistorySave(const char *filename) {
     fp = fopen(filename,"w");
     umask(old_umask);
     if (fp == NULL) return -1;
-    chmod(filename,S_IRUSR|S_IWUSR);
+    if (chmod(filename,S_IRUSR|S_IWUSR) < 0) {
```

Comment:
So what happens in the very odd case where we can get a file handle but can't immediately run a system call on it? 

I would however cast the return value (void) as a reminder we didn't need the return value.

Reply:
As I commented on the referenced commit, why have the chmod at all?